lib/workspace.rb

```diff
@@ -0,0 +1,63 @@
+require "pry"
+require "json"
+
+require_relative "user"
+require_relative "channel"
+
+class Workspace
+  attr_reader :users, :channels, :selected
+
+  def initialize
+    @users = User.list
+    @channels = Channel.list
+    @selected = nil
+  end
+
+  def select_channel(user_input)
+    selected = channels.select do |channel|
+      channel.name == user_input || channel.slack_id == user_input
+    end
+    @selected = selected.first
+  end
+
+  def select_user(user_input)
+    selected = users.select do |user|
+      user.name == user_input || user.slack_id == user_input
+    end
+    @selected = selected.first
+  end
+
+  def show_details
+    return @selected.details
+  end
+
+  def print_details(recipients)
+    if recipients == "users"
```

Comment:
I'd like to see `recipients` just be an array of objects, but if you want it to be something you switch on like this you should use `:users` instead of `"users"` since generally we want to use atoms when we match on things like this.